libraries/lib-ipc/internal/BufferedIPCChannel.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <mutex>
+#include <condition_variable>
+#include <memory>
+#include <vector>
+#include <thread>
+
+#include "ipc-types.h"
+#include "IPCChannel.h"
+
+class IPCChannelStatusCallback;
+
+/**
+ * \brief Socket-based implementation of IPCChannel that uses intermediate
+ * buffer for data exchange between client and server.
+ */
+class BufferedIPCChannel final : public IPCChannel
+{
+   static constexpr int DefaultOutputBufferCapacity  { 2048 };
+   static constexpr int DefaultInputBufferSize { 2048 };
+
+   bool mAlive {true};
+   std::mutex mSocketSync;
+   std::condition_variable mSendCondition;
+
+   std::unique_ptr<std::thread> mRecvRoutine;
+   std::unique_ptr<std::thread> mSendRoutine;
```

Comment:
`unique_ptr`s to Threads again... I still say it is an unnecessary indirection and you should test with `joinable` whether a thread is really there.